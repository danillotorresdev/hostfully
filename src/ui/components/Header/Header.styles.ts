import { styled } from "@linaria/react";

export const Header = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: var(--color-primary);
  color: var(--color-white);
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1);
  flex-direction: column;

  img {
    padding-bottom: 1rem;
  }

  @media (min-width: 768px) {
    flex-direction: row;
    img {
      padding-bottom: 0;
    }
  }
`;

export const Nav = styled.nav`
  display: flex;
  gap: 2rem;
`;

export const NavLink = styled.a`
  color: var(--color-white);
  text-decoration: none;
  font-size: 1rem;
  font-weight: 500;
`;
