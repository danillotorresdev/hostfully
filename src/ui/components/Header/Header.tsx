import { Logo } from "@/ui/components/Logo/Logo";
import * as S from "./Header.styles";

export const HeaderComponent = () => {
  return (
    <S.Header>
      <Logo />
      <S.Nav>
        <S.NavLink href="#">Home</S.NavLink>
        <S.NavLink href="#">About</S.NavLink>
        <S.NavLink href="#">Contact</S.NavLink>
      </S.Nav>
    </S.Header>
  );
};
