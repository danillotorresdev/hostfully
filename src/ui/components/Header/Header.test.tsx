import { HeaderComponent } from "./Header";
import { describe, expect, it } from "vitest";

import { render, screen } from "@/utils/settings/tests/utilities";

describe("Header", () => {
  it("should render the header", () => {
    const { asFragment } = render(<HeaderComponent />);

    const homeAnchor = screen.getByText("Home");
    const aboutAnchor = screen.getByText("About");
    const contactAnchor = screen.getByText("Contact");

    expect(asFragment()).toMatchSnapshot();
    expect(homeAnchor).toBeDefined();
    expect(aboutAnchor).toBeDefined();
    expect(contactAnchor).toBeDefined();
  });
});
