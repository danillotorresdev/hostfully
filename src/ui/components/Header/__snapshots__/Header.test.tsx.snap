// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render the header 1`] = `
<DocumentFragment>
  <header
    class="mocked-styled-1 hl4mxfd"
  >
    <img
      alt="Logo"
      class="mocked-styled-0 le32cth"
      src="https://www.hostfully.com/wp-content/uploads/2022/08/logo-1.svg"
      style="width: 200px; height: auto; display: block;"
    />
    <nav
      class="mocked-styled-2 n2miw7l"
    >
      <a
        class="mocked-styled-3 n3754ll"
        href="#"
      >
        Home
      </a>
      <a
        class="mocked-styled-3 n3754ll"
        href="#"
      >
        About
      </a>
      <a
        class="mocked-styled-3 n3754ll"
        href="#"
      >
        Contact
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
